tests/e2e/vmservice_vm/rwx_vmsvc_utils.go

```diff
@@ -0,0 +1,204 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vmservice_vm
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	vsantypes "github.com/vmware/govmomi/vsan/types"
+	pkgtypes "k8s.io/apimachinery/pkg/types"
+	cnsfileaccessconfigv1alpha1 "sigs.k8s.io/vsphere-csi-driver/v3/pkg/apis/cnsoperator/cnsfileaccessconfig/v1alpha1"
+	"sigs.k8s.io/vsphere-csi-driver/v3/tests/e2e/constants"
+	"sigs.k8s.io/vsphere-csi-driver/v3/tests/e2e/k8testutil"
+	"sigs.k8s.io/vsphere-csi-driver/v3/tests/e2e/vcutil"
+)
+
+// CreateCnsFileAccessConfigCRD creates CnsFileAccessConfigCRD using pvc and VMservice VM name
+// in a given namespace
+func CreateCnsFileAccessConfigCRD(ctx context.Context, restConfig *rest.Config, pvcName string,
+	vmsvcVmName string, namespace string, crdName string) error {
+	cnsOperatorClient, err := vcutil.GetCnsOperatorClient(ctx, restConfig)
+	if err != nil {
+		return err
+	}
+	spec := &cnsfileaccessconfigv1alpha1.CnsFileAccessConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      crdName,
+			Namespace: namespace},
+		Spec: cnsfileaccessconfigv1alpha1.CnsFileAccessConfigSpec{
+			VMName:  vmsvcVmName,
+			PvcName: pvcName,
+		},
+	}
+
+	if err := cnsOperatorClient.Create(ctx, spec); err != nil {
+		return fmt.Errorf("failed to create CNSFileAccessConfig CRD: %s with err: %v", crdName, err)
+	}
+	return nil
+}
+
+// FetchNFSAccessPointFromCnsFileAccessConfigCRD returns a NFS access point from CnsFileAccessConfigCRD
+func FetchNFSAccessPointFromCnsFileAccessConfigCRD(ctx context.Context, restConfig *rest.Config,
+	cnsFileAccessConfigCRDName string, namespace string) (string, error) {
+	cfc, err := GetCnsFileAccessConfigCRD(ctx, restConfig, cnsFileAccessConfigCRDName, namespace)
+	if err != nil {
+		return "", err
+	}
+	nfs4AccessPoint := cfc.Status.AccessPoints[constants.Nfs4Keyword]
+	return nfs4AccessPoint, nil
+}
+
+// Helper function to run SSH and log result
+func RunSSHFromVmServiceVmAndLog(vmIP, cmd string) error {
+	output := execSshOnVmThroughGatewayVm(vmIP, []string{cmd})
+	if output[0].Stderr != "" {
+		return fmt.Errorf("Command failed with error: %s", output[0].Stderr)
+	} else {
+		framework.Logf("Executed on %s: %s\nOutput: %v", vmIP, cmd, output[0].Stdout)
+		return nil
+	}
+}
+
+// MountRWXVolumeAndVerifyIO mounts RWX volume inside a given VMService VM using the NFS Access point
+func MountRWXVolumeAndVerifyIO(vmIPs []string, nfsAccessPoint string, testDir string) error {
```

Comment:
take nfsAccessPoint as an array in case multplie rwx attached to vm

Reply:
Will take up in next PR